core/deployment/src/main/java/io/quarkus/deployment/steps/ConfigGenerationBuildStep.java

```diff
@@ -204,20 +204,27 @@ void generateMappings(
             BuildProducer<ConfigClassBuildItem> configClasses,
             BuildProducer<AdditionalConstrainedClassBuildItem> additionalConstrainedClasses) {
 
-        processConfigMapping(combinedIndex, generatedClasses, reflectiveClasses, reflectiveMethods, configClasses,
+        Map<String, GeneratedClassBuildItem> generatedConfigClasses = new HashMap<>();
```

Comment:
(Not very familiar with the code-generation code, so ignore my comment, if it doesn't apply)

Should this be a `LinkedHashMap` for reproducibility?